### PR TITLE
Add CI builds, add owasp profile

### PR DESCRIPTION
**Pull request by oconnormi**
### Description of the Change

Adds Jenkins pipeline support:
* Will comment on pr at beginning and end of build process
* Performs Sonar scans

### Verification Processn
Run `Build Now`

### Applicable Issues

Fixes: #3 

### Remaining Tasks:
- [x] Create sonar project

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        /*
+          Restrict nightly builds to master branch, all others will be built on change only.
+          Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
+        */
+        cron(BRANCH_NAME == "master" ? "H H(17-19) * * *" : "")
+    }
+    environment {
+        DISABLE_DOWNLOAD_PROGRESS_OPTS = '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
+        COVERAGE_EXCLUSIONS = '**/test/**/*,**/itests/**/*,**/*Test*,**/sdk/**/*,**/*.js,**/node_modules/**/*,**/jaxb/**/*,**/wsdl/**/*,**/nces/sws/**/*,**/*.adoc,**/*.txt,**/*.xml,**/platform-solr-server-standalone/**/*'
```
Comment (paouelle) [inline Jenkinsfile]: ❓ Seems like a cut and paste. Should it be sanitize for this repo?

**Review by paouelle — APPROVED** [review]
Approved with one question

**Review by coyotesqrl — APPROVED** [review]

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
Got a strange error when the windows build tried to clone the repo, trying again. Build Now

**Comment by oconnormi** [comment]
Assuming we want OWASP on this one, the pipeline itself is ready to go. There are several OWASP failures that are currently failing that stage:

```bash
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:3.1.1:check (default) on project jacoco: 

[ERROR] 

[ERROR] One or more dependencies were identified with vulnerabilities that have a CVSS score greater than or equal to '2.0': 

[ERROR] 

[ERROR] xercesImpl-2.8.1.jar: CVE-2012-0881

[ERROR] commons-collections-3.2.jar: CVE-2015-6420, CVE-2017-15708

[ERROR] commons-beanutils-1.7.0.jar: CVE-2014-0114
```

@paouelle, @coyotesqrl do we need to do anything about these failures, or are we suppressing them?

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
Build Now

**Pull request merged**